### ISSUE TITLE
Node with block list property failed to be shown after upgrading project from U13.3.0 to U14-rc1

### ISSUE THREAD

**Issue by dydimitrov** [umbraco/Umbraco-CMS#16103]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

14.0.0-rc2

### Bug summary

After upgrading the project from Umbraco 13.3.0 to Umbraco 14.0.0-rc2 I have nodes where I have only one property with a block list editor but when I try to access the node I get an error saying "The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. Path: $ | LineNumber: 0 | BytePositionInLine: 0."
[JSON-Tokens-Error.md](https://github.com/umbraco/Umbraco-CMS/files/15036327/JSON-Tokens-Error.md)
<img width="956" alt="U14-RC2-JSON-Tokens-Error" src="https://github.com/umbraco/Umbraco-CMS/assets/22914054/f92f222f-b666-4550-8d27-ebee39a3caf3">
a 

### Steps to reproduce

Upgrade Existing project from U 13.3.0 to Umbraco 14.0.0-rc2
Try to open Node where Block list editor property is used

### Expected result / actual result

To show the node with the property editor together with the items inside

**Comment by elit0451** [comment]
Hi there @dydimitrov  👋

Thanks for reaching out! I didn't get the same error as you - I managed to open the node with the Block list - but I can see an error in the logs from clicking on the items. 

![BLE](https://github.com/umbraco/Umbraco-CMS/assets/21998037/1e7e5974-7cb0-4c71-a82a-df3deb51dc07)

Plus the Block list data wasn't migrated. Here is the error I got for reference:

```
DISTRIBUTED CACHE IS NOT UPDATED. Failed to execute instructions ...

System.ArgumentNullException: Value cannot be null. (Parameter 'key')
   at System.Collections.Generic.Dictionary`2.FindValue(TKey key)
   at System.Collections.Generic.Dictionary`2.TryGetValue(TKey key, TValue& value)
   at Umbraco.Cms.Infrastructure.Search.ContentTypeIndexingNotificationHandler.Handle(ContentTypeCacheRefresherNotification args) in /UmbracoCMS/src/Umbraco.Infrastructure/Search/IndexingNotificationHandler.ContentType.cs:line 60
   at Umbraco.Cms.Core.Events.INotificationHandler`1.Handle(IEnumerable`1 notifications) in /UmbracoCMS/src/Umbraco.Core/Events/INotificationHandler.cs:line 35
   at Umbraco.Cms.Core.Events.NotificationHandlerWrapperImpl`1.<>c__DisplayClass0_0`2.<Handle>b__2(IEnumerable`1 handlerNotifications) in /UmbracoCMS/src/Umbraco.Core/Events/EventAggregator.Notifications.cs:line 284
   at Umbraco.Cms.Core.Events.EventAggregator.PublishCore[TNotification](IEnumerable`1 allHandlers, IEnumerable`1 notifications) in /UmbracoCMS/src/Umbraco.Core/Events/EventAggregator.Notifications.cs:line 109
   at Umbraco.Cms.Core.Events.NotificationHandlerWrapperImpl`1.Handle[TNotification,TNotificationHandler](IEnumerable`1 notifications, ServiceFactory serviceFactory, Action`2 publish) in /UmbracoCMS/src/Umbraco.Core/Events/EventAggregator.Notifications.cs:line 286
   at Umbraco.Cms.Core.Events.EventAggregator.PublishNotifications[TNotification,TNotificationHandler](IEnumerable`1 notifications) in /UmbracoCMS/src/Umbraco.Core/Events/EventAggregator.Notifications.cs:line 77
   at Umbraco.Cms.Core.Events.EventAggregator.Publish[TNotification,TNotificationHandler](IEnumerable`1 notifications) in /UmbracoCMS/src/Umbraco.Core/Events/EventAggregator.cs:line 26
   at Umbraco.Cms.Core.Events.EventAggregator.Publish[TNotification](TNotification notification) in /UmbracoCMS/src/Umbraco.Core/Events/EventAggregator.cs:line 18
   at Umbraco.Cms.Core.Cache.CacheRefresherBase`1.OnCacheUpdated(CacheRefresherNotification notification) in /UmbracoCMS/src/Umbraco.Core/Cache/Refreshers/CacheRefresherBase.cs:line 101
   at Umbraco.Cms.Core.Cache.PayloadCacheRefresherBase`2.Refresh(TPayload[] payloads) in /UmbracoCMS/src/Umbraco.Core/Cache/Refreshers/PayloadCacheRefresherBase.cs:line 47
   at Umbraco.Cms.Core.Cache.ContentTypeCacheRefresher.Refresh(JsonPayload[] payloads) in /UmbracoCMS/src/Umbraco.Core/Cache/Refreshers/Implement/ContentTypeCacheRefresher.cs:line 123
   at Umbraco.Cms.Core.Cache.PayloadCacheRefresherBase`2.Refresh(String json) in /UmbracoCMS/src/Umbraco.Core/Cache/Refreshers/PayloadCacheRefresherBase.cs:line 38
   at Umbraco.Cms.Infrastructure.Services.CacheInstructionService.RefreshByJson(CacheRefresherCollection cacheRefreshers, Guid uniqueIdentifier, String jsonPayload) in /UmbracoCMS/src/Umbraco.Infrastructure/Services/CacheInstructionService.cs:line 455
   at Umbraco.Cms.Infrastructure.Services.CacheInstructionService.NotifyRefreshers(CacheRefresherCollection cacheRefreshers, IEnumerable`1 instructions, HashSet`1 processed, CancellationToken cancellationToken) in /UmbracoCMS/src/Umbraco.Infrastructure/Services/CacheInstructionService.cs:line 401
   at Umbraco.Cms.Infrastructure.Services.CacheInstructionService.ProcessDatabaseInstructions(CacheRefresherCollection cacheRefreshers, IReadOnlyCollection`1 instructionBatch, CacheInstruction instruction, HashSet`1 processed, CancellationToken cancellationToken, Int32& lastId) in /UmbracoCMS/src/Umbraco.Infrastructure/Services/CacheInstructionService.cs:line 335
```


We will have a look 🙂

**Comment by dydimitrov** [comment]
Hi @elit0451 
I would assume then that it is because I have dropdown and checkboxes in the block list editor items, also I have at least 2 levels of block lists. I mean I have the direct property under the node let's say BLE_Level1 --> BLE_level2--> Item containing dropdown or checkbox. I hope that this will help.

**Comment by iOvergaard** [comment]
@dydimitrov A lot of migrations got fixed just before the launch of 14.0 and a few more with 14.1. Could I persuade you to check this again with 14.1?

**Comment by nielslyngsoe** [comment]
Hi @dydimitrov Block List and Block Grid Editor was one of the few things that was not fully working in the first releases of v.14 — Could you please reply back if you still experience this issue with later versions.

Until then I will close of the issue, if so please re-open it. Thanks in advance
